js/script.js

```diff
@@ -0,0 +1,49 @@
+$(document).ready(function(){
+	var images = []
+	var interval = -1
+	var index = 0
+
+ 	function getImages(subject) {
+ 		var url = "http://www.reddit.com/search.json?q="+subject+"nsfw:no";
+ 		$.get(url, {
+ 			}).done(function(data){
+ 			 images = data.data.children.map(function(child){
+ 				return child.data;
+ 				}).filter(function(child) {
+ 					return child.preview != undefined;
+ 				}).map(function(child) {
+ 					return child.preview.images[0].source.url
+ 				})
+ 				console.log(images);
+ 			});
+ 			var filtered = searchResult.filter(function (filter){
+ 				return filter !== undefined;
+ 			});
+
+	getImages("cat");
+
+	function startSlideshow() {
+		if (interval != -1) {
+			stopSlideshow();
+		}
+	}
+	interval = setInterval(function() {
```

Comment:
this `setInterval` should be inside a function and then called after the search.